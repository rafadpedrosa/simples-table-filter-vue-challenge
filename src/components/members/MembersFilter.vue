<template>
<div class="d-flex justify-content-between">
  <b-form @submit="event.preventDefault()" inline>
      <b-form-group
        label="ROLE"
        class="mb-3 mr-sm-3 mb-sm-0"
      >
        <b-form-select
          :options="[{ text: 'All Roles', value: 'All' }, 'Owner', 'Viewer', 'Editor']"
          :value="'All'"
        ></b-form-select>
      </b-form-group>
      <b-form-group
        label="TIME TRACKING"
      >
        <b-form-select
          :options="[{ text: 'Enabled', value: true },{ text: 'Enabled', value: false }]"
          :value="true"
        ></b-form-select>
      </b-form-group>
  </b-form>

  <b-button variant="primary" class="align-self-end" @click="console.log(':::not implemented')"> Invite member </b-button>
</div>
</template>

<script>
export default {
  name: 'MembersFilter',
  data () {
    return {
      form: {
        role: '',
        timeTrack: ''
      }
    }
  }
}
</script>

<style scoped>

</style>
