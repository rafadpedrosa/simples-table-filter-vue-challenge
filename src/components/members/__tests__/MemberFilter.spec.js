import { mount } from '@vue/test-utils'
import MembersFilter from '@/components/members/MembersFilter'

/**
 * COMPONENT CONTRACT
 *
 * 1- It Should be rendered with the all roles and enabled already selected
 * 2- It should emit an event called filters when the data is changed
 **/

const EMPTY_STRING = ''
const ENABLED = 'enabled'

describe('MembersFilter.vue', () => {
  let wrapper

  beforeAll(() => {
    wrapper = mount(MembersFilter, {})
  })

  it('should be rendered with the all roles and enabled already selected', () => {
    const filters = wrapper.vm.filters

    expect(wrapper).toMatchSnapshot()
    expect(filters.role).toBe(EMPTY_STRING)
    expect(filters.timeTrack).toBe(ENABLED)
  })

  it('should emit an event called filters when the data is changed', async () => {
    const roleElement = wrapper.find('select[data-test=role]').findAll('option')
    const timeTrackElement = wrapper.find('select[data-test=timeTrack]').findAll('option')

    const { role, timeTrack } = wrapper.vm.filters

    await roleElement.at(2).setSelected()
    await timeTrackElement.at(1).setSelected()

    const { role: newRole, timeTrack: newTimeTrack } = wrapper.vm.filters

    expect(role).not.toBe(newRole)
    expect(timeTrack).not.toBe(newTimeTrack)
    expect(wrapper.emitted().filters).toHaveLength(2)
    expect(wrapper).toMatchSnapshot()
  })
})
