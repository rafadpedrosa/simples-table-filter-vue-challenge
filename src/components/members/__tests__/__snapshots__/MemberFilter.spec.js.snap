// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MembersFilter.vue should be rendered with the all roles and enabled already selected 1`] = `
<div class="d-flex justify-content-between">
  <form class="form-inline">
    <fieldset class="form-group mb-3 mr-sm-3 mb-sm-0" id="__BVID__4">
      <legend tabindex="-1" class="bv-no-focus-ring col-form-label pt-0" id="__BVID__4__BV_label_">ROLE</legend>
      <div tabindex="-1" role="group" class="bv-no-focus-ring"><select class="custom-select" data-test="role" id="__BVID__5">
          <option value="">All Roles</option>
          <option value="Owner">Owner</option>
          <option value="Viewer">Viewer</option>
          <option value="Editor">Editor</option>
        </select>
        <!---->
        <!---->
        <!---->
      </div>
    </fieldset>
    <fieldset class="form-group" id="__BVID__6">
      <legend tabindex="-1" class="bv-no-focus-ring col-form-label pt-0" id="__BVID__6__BV_label_">TIME TRACKING</legend>
      <div tabindex="-1" role="group" class="bv-no-focus-ring"><select class="custom-select" data-test="timeTrack" id="__BVID__7">
          <option value="enabled">Enabled</option>
          <option value="disabled">Disable</option>
        </select>
        <!---->
        <!---->
        <!---->
      </div>
    </fieldset>
  </form> <button type="button" class="btn align-self-end btn-primary"> Invite member </button>
</div>
`;

exports[`MembersFilter.vue should emit an event called filters when the data is changed 1`] = `
<div class="d-flex justify-content-between">
  <form class="form-inline">
    <fieldset class="form-group mb-3 mr-sm-3 mb-sm-0" id="__BVID__4">
      <legend tabindex="-1" class="bv-no-focus-ring col-form-label pt-0" id="__BVID__4__BV_label_">ROLE</legend>
      <div tabindex="-1" role="group" class="bv-no-focus-ring"><select class="custom-select" data-test="role" id="__BVID__5">
          <option value="">All Roles</option>
          <option value="Owner">Owner</option>
          <option value="Viewer">Viewer</option>
          <option value="Editor">Editor</option>
        </select>
        <!---->
        <!---->
        <!---->
      </div>
    </fieldset>
    <fieldset class="form-group" id="__BVID__6">
      <legend tabindex="-1" class="bv-no-focus-ring col-form-label pt-0" id="__BVID__6__BV_label_">TIME TRACKING</legend>
      <div tabindex="-1" role="group" class="bv-no-focus-ring"><select class="custom-select" data-test="timeTrack" id="__BVID__7">
          <option value="enabled">Enabled</option>
          <option value="disabled">Disable</option>
        </select>
        <!---->
        <!---->
        <!---->
      </div>
    </fieldset>
  </form> <button type="button" class="btn align-self-end btn-primary"> Invite member </button>
</div>
`;
